### PR TITLE
Added steps to make CI works for a specific patch

### PR DESCRIPTION
**Pull request by praveenkumar**
Till now we were using master tarball for generating source rpm which was blocking us to use a specific PR in our CI for ondemand use case. This patch will directly create tarball from source.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -26,9 +25,16 @@ DOWNSTREAM_IMAGE_VERSION := "v3\.2\.0\.20"
...
 upstream:
 
-	curl -sL -O ${SOURCE}
+	sed -i ${UPSTREAM_NAME}.spec \
+	    -e "s|^Source0:\(\s\+\)\(.*\)$ |Source0:\1\%{name}-\%{version}\.tar\.gz|" \
+	    -e "s|^Version:\(\s\+\)\(.*\)$ |Version:\1${VERSION}|"
+	mkdir -p ${UPSTREAM_NAME}-${VERSION}
+	tar cf - --exclude=${UPSTREAM_NAME}-${VERSION} --exclude=.git . | (cd ${UPSTREAM_NAME}-${VERSION} && tar xvf - )
+	tar -cvf ${UPSTREAM_NAME}-${VERSION}.tar.gz ${UPSTREAM_NAME}-${VERSION} && \
+	    rm -fr ${UPSTREAM_NAME}-${VERSION}
```
Comment (LalatenduMohanty) [inline Makefile]: @praveenkumar I am having hard time understanding the sed commands and their purpose. Can these be simplified or can we add some comments which will help understand them?

**Comment by cgwalters** [comment]
FWIW https://github.com/cgwalters/rpmdistro-gitoverlay is a tool like this and I'm actively working on integrating it with PR testing for things in Atomic Host in CentOS CI:

https://github.com/cgwalters/sig-atomic-buildscripts/blob/pr-testing/centos-ci/jjb/sig-atomic-testing.yml#L44

**Comment by praveenkumar** [comment]
@cgwalters Thanks to pointing out this tool, currently we are solving problem downstream mainly and we have plan for add CI for upstream (that case we will follow your job-builder files and have close look of rpmdistro-gitoverlay).

**Comment by praveenkumar** [comment]
@LalatenduMohanty I used `git archive` in upstream but in `master` which resemble with downstream latest there `tar` is what good way to handle it because we do make some modification in files locally even for PR we require to do that.

**Comment by LalatenduMohanty** [comment]
LGTM

**Comment by LalatenduMohanty** [comment]
@cgwalters Thanks for the pointer to https://github.com/cgwalters/rpmdistro-gitoverlay.

**Pull request merged**